File: easy/minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.py
from itertools import accumulate


class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        startVal, s = 1, 0
        for n in nums:
            diff = n + s
            if diff < 0:
                startVal -= diff
                s -= diff
            s += n
        return startVal


class Solution2:
    def minStartValue(self, nums: List[int]) -> int:
        return max(1, -min(accumulate(nums)) + 1)

File: easy/minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.rs
impl Solution {
  pub fn min_start_value(nums: Vec<i32>) -> i32 {
    let mut start = 1;
    let mut sum = 0;
    for n in nums.iter() {
      let diff = n + sum;
      if diff < 0 {
        start -= diff;
        sum -= diff;
      }
      sum += n;
    }
    start
  }
}

struct Solution {}

fn main() {
  assert_eq!(0, 0);
}
